app/assets/javascripts/youtube.coffee

```diff
@@ -0,0 +1,49 @@
+$ ->
+  widget = ''
+  player = ''
+
+  onUploadSuccess = (event) ->
+    $('#uploading_in_progress').show()
+    $.ajax(
+      type: 'POST',
+      url: httpRoot + 'videos',
+      data:
+        video:
+          youtube_id: event.data.videoId
+      success: ->
+        console.log('Worked!')
+      error: (jqxhr, textStatus, errorThrown) ->
+        console.log('jqxhr:', jqxhr)
+        console.log('textStatus:', textStatus)
+        console.log('errorThrown:', errorThrown)
+    )
+
+  onProcessingComplete = (event) ->
+    $('#uploading_in_progress').hide()
+    player = new YT.Player(
+      'player',
+      height: 390,
+      width: 640,
+      videoId: event.data.videoId,
+      events: {}
+    )
+
+  if $('.videos-index span.video').length > 0
+    window.onYouTubeIframeAPIReady = ->
```

Comment:
The async youtube framework looks for a global function with a specific name to fire when it's loaded.